httpcore5-h2/src/main/java/org/apache/hc/core5/http2/impl/nio/AbstractHttp2StreamMultiplexer.java

```diff
@@ -610,7 +612,7 @@ private void processPendingCommands() throws IOException, HttpException {
                         connMetrics,
                         exchangeHandler,
                         context);
-                final Http2Stream stream = new Http2Stream(channel, streamHandler, false);
+                final Http2Stream stream = new Http2Stream(channel, streamHandler, pushConsumer, false);
```

Comment:
We must stash the push consumer on some object that's attached to the original stream so we can later get at it when creating the new promised stream. I could see at least three ways of doing that:
- on the stream directly (as done here)
- on the stream handler
- adding an 'acceptPromise' method to the exchange handler.

I must had admit I was tempted to modify the exchange handler. But it didn't fit with your API proposal and I'm not confident enough with the code base to understand the design consequences.

Reply:
@ato Feel free to do what you deem right / proper. My proposal was just a quick sketch. Feel free to disregard it.

Reply:
Thanks. In that case I will close this for now and try the other approach.